crates/orchestrator/src/network/node.rs

```diff
@@ -37,6 +37,58 @@ pub struct NetworkNode {
     metrics_cache: Arc<RwLock<MetricMap>>,
 }
 
+/// Result of waiting for a certain number of log lines to appear.
+///
+/// Indicates whether the log line count condition was met within the timeout period.
+///
+/// # Variants
+/// - `TargetReached(count)` – The predicate condition was satisfied within the timeout.
+///     * `count`: The number of matching log lines at the time of satisfaction.
+/// - `TargetFailed(count)` – The condition was not met within the timeout.
+///     * `count`: The final number of matching log lines at timeout expiration.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum LogLineCount {
```

Comment:
Random idea: WDYT about having a `target_reached` or `success` method. This would make it easier to to just assert success without needing to use `matches` like in your breaking changes PR.

Reply:
Ha, actually I really like this idea. Will add it!

Reply:
Done